### PR TITLE
[DPE-3926] Enforce zookeeper client interface

### PR DESCRIPTION
**Pull request by Batalex**
This PR adds `database` and `requested-secrets` to the relation databag, while preserving the `chroot` entry for compatibility with older zookeeper charms

### CODE REVIEW

File: src/events/zookeeper.py [hunk 1]
```diff
@@ -38,7 +38,13 @@ def __init__(self, charm) -> None:
     def _on_zookeeper_created(self, _) -> None:
         """Handler for `zookeeper_relation_created` events."""
         if self.model.unit.is_leader():
-            self.charm.state.zookeeper.update({"chroot": "/" + self.model.app.name})
+            self.charm.state.zookeeper.update(
+                {
+                    "database": "/" + self.model.app.name,
+                    "requested-secrets": '["username","password","tls","tls-ca","uris"]',
+                    "chroot": "/" + self.model.app.name,
```
Comment (deusebio) [inline src/events/zookeeper.py:45]: **thought** I'm actually wondering whether we could avoid to keep on setting `chroot`. This is only created when the relation is established with zookeeper, so it is really not that we break upgrades or existing deployment (as long as in existing deployments we are robust to this field), but rather that from now and on we requires ZooKeeper to have secrets enabled when establishing the relation. 

Of course, if we drop this field, Kafka <> ZooKeeper relation creation would not work when Kafka with Secrets tries to related to ZooKeeper prior of secrets introduction, but I'm actually wondering whether this is really important to support, and worth for having the databag a bit dirtier. 

Also if we want to keep it for the time being, what would be the way to sunset its usage or are we going to keep it forever?
  backward compatibility. , and require in the dependencies for the upgrade to only have

Comment (Batalex) [reply src/events/zookeeper.py:45]: Having old kafka <-> zk and old zk <-> kafka working together was a requirement. This also made sense in the context where the introduction of secrets was done in a backward compatible way, with an abstraction layer that converts secrets to normal fields if needed and vice versa.

I am +1 on defining a deprecation policy

Comment (deusebio) [reply src/events/zookeeper.py:45]: I have checked this with Mykola, we can skip this field and break backward compatibility when we will migrate to a major version, say on Kafka 4/.

**Review by deusebio — APPROVED** [review]

**Review by zmraul — APPROVED** [review]
LGTM!

**Pull request merged**